subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -266,6 +288,51 @@ static int cmd_wifi_scan(const struct shell *shell, size_t argc, char *argv[])
 	return 0;
 }
 
+static int cmd_wifi_status(const struct shell *shell, size_t argc, char *argv[])
+{
+	struct net_if *iface = net_if_get_default();
+	struct wifi_iface_status status = {0,};
+
+	context.shell = shell;
+
+	if (net_mgmt(NET_REQUEST_WIFI_IFACE_STATUS, iface, &status,
+				sizeof(struct wifi_iface_status))) {
+		shell_fprintf(shell, SHELL_WARNING, "Status request failed\n");
+
+		return -ENOEXEC;
+	}
+
+	shell_fprintf(shell, SHELL_NORMAL, "Status: successful\n");
+	shell_fprintf(shell, SHELL_NORMAL, "==================\n");
+	shell_fprintf(shell, SHELL_NORMAL, "State: %s\n", wifi_state_txt(status.state));
+
+	if (status.state >= WIFI_ASSOCIATED) {
+		uint8_t mac_string_buf[sizeof("xx:xx:xx:xx:xx:xx")];
+
+		shell_fprintf(shell, SHELL_NORMAL, "Interface Mode: %s\n",
+				wifi_mode_txt(status.iface_mode));
+		shell_fprintf(shell, SHELL_NORMAL, "Link Mode: %s\n",
+				wifi_link_mode_txt(status.link_mode));
+		shell_fprintf(shell, SHELL_NORMAL, "SSID: %-32s\n", status.ssid);
+		shell_fprintf(shell, SHELL_NORMAL, "BSSID: %s\n",
+				status.bssid ? net_sprint_ll_addr_buf(status.bssid,
+					WIFI_MAC_ADDR_LEN, mac_string_buf,
+					sizeof(mac_string_buf)) :
+				"");
+		shell_fprintf(shell, SHELL_NORMAL, "Band: %s\n",
+				wifi_band_txt(status.band));
+		shell_fprintf(shell, SHELL_NORMAL, "Channel: %d\n", status.channel);
+		shell_fprintf(shell, SHELL_NORMAL, "Security: %s\n",
+				wifi_security_txt(status.security));
+		shell_fprintf(shell, SHELL_NORMAL, "PMF: %s\n",
```

Comment:
s/PMF/MFP/ ?

Reply:
This was intentional, MFP is the IEEE name, but PMF is the most commonly used name, so, using PMF for public.

Reply:
But this is basically the same thing? Won't it be confusing if we mix those two? Usually it is better to stick with one, so that `grep` does the job without the requirement to have context of what PMF/MFP means (e.g. I don't know what it means).

Reply:
I was just following wpa_supplicant style (https://w1.fi/wpa_supplicant/devel/structwpa__config.html#adc14d004b51292339ada9ed6f5d276b8), but sure we can stick to MFP.

MFP - Management frame protection
PMF - Protected Management frames

Reply:
Personally I would prefer to use one or the other, I feel like it would be easier to jump in for newcomers (like me in case of WiFi). But don't treat that as hard requirement from me.

I didn't work much with `wpa_supplicant`, but heard a lot about legacy that comes with it. I think that using one name for a single thing should be the way to go in our codebase. If needed, we can use both names in the glue layer between wpa_supplicant and Zephyr.